backend/app/repositories/search_profile_repository.py

```diff
@@ -1,113 +1,151 @@
-
-            if data.profile_emails is not None:
-                profile.profile_emails = data.profile_emails
-
-            # Update subscriptions (replace with new list)
-            if data.subscriptions is not None:
-                profile.subscriptions.clear()
-                for sub_data in data.subscriptions:
-                    profile.subscriptions.append(sub_data.to_model())
-
-            # Update topics (replace with new list)
-            if data.topics is not None:
-                profile.topics.clear()
-                for topic_data in data.topics:
-                    profile.topics.append(topic_data.to_model())
-
-            session.add(profile)
-            await session.commit()
-            await session.refresh(profile)
-
-            return profile
+        )
+
+    return response_profiles
+
+
+async def update_by_id(
+    search_profile_id: UUID, data: SearchProfileUpdateRequest
+) -> Type[SearchProfile] | None:
+    async with async_session() as session:
+        profile = await session.get(SearchProfile, search_profile_id)
+        if not profile:
+            return None
+
+        # Update simple fields
+        profile.name = data.name
+        profile.is_public = data.public
+        profile.is_editable = data.is_editable
+        profile.owner_id = data.owner
+        profile.is_owner = data.is_owner
+
+        # Update email lists
+        if data.organization_emails is not None:
+            profile.organization_emails = data.organization_emails
+
+        if data.profile_emails is not None:
+            profile.profile_emails = data.profile_emails
+
+        # Update subscriptions (replace with new list)
+        if data.subscriptions is not None:
+            profile.subscriptions.clear()
+            for sub_data in data.subscriptions:
+                profile.subscriptions.append(sub_data.to_model())
+
+        # Update topics (replace with new list)
+        if data.topics is not None:
+            profile.topics.clear()
+            for topic_data in data.topics:
+                profile.topics.append(topic_data.to_model())
+
+        session.add(profile)
+        await session.commit()
+        await session.refresh(profile)
+
+        return profile
+
+
+async def get_accessible_profiles(user_id, organization_id):
+    async with async_session() as session:
+        result = await session.execute(
+            select(SearchProfile)
+            .options(
+                selectinload(SearchProfile.users),
+                selectinload(SearchProfile.topics).selectinload(
+                    Topic.keywords
+                ),
+            )
+            .where(
+                SearchProfile.is_public
+                | (SearchProfile.organization_id == organization_id)
+                | (SearchProfile.users.any(User.id == user_id))
+            )
+        )
+        return result.unique().scalars().all()
+
+
+async def get_by_id(
+    search_profile_id: UUID, current_user: User
```

Comment:
```suggestion
    search_profile_id: UUID
```

you're not using this user. can it be removed or did you forget to check if the user has access to the search profile here?